### DISCUSSION TITLE
How does the equilibrium point of the 6DOFJoint of the PhysicalBone3D work?

### DISCUSSION THREAD

**Discussion by V1LLz** [godot-jolt/godot-jolt#826] [Questions]
Hello everyone, I had a question regarding the operation of the equilibrium point in the 6DOFJoint of PhysicalBone3D. I have tested this joint with rigid bodies and static bodies, and the angle of the equilibrium point corresponds to the angular limits and is shown in degrees. However, when you set these angular limits in a PhysicalBone3D and try to modify the equilibrium point, it is not displayed in degrees in the inspector. Instead, it shows decimals that, from what I have seen, are close to the value of π, which makes it a bit complicated to set them, either via code or inspector, as you don't know how far the angular limits of that bone go. I'm not sure if this is specific to Godot or if it's some kind of error.

The other question is a query again regarding the topic, and it's whether there are plans to add the missing motor functions and additional functions (softness, restitution, damping, ERP) to the 6DOFJoint of PhysicalBone3D at some point. That's all, thank you very much for reading! 😄

**Comment by mihe** [comment]
> I'm not sure if this is specific to Godot or if it's some kind of error.

This is indeed a bug in Godot, and not something I have control over from this extension. You are currently forced to type in radians as opposed to degrees. [This](https://github.com/godotengine/godot/blob/97b5c142f8f90a178c64d89e8e4ea4cd330d0b4d/scene/3d/physics/physical_bone_3d.cpp#L703) line probably needs to be:

```diff
-p_list->push_back(PropertyInfo(Variant::FLOAT, prefix + PNAME("angular_equilibrium_point")));
+p_list->push_back(PropertyInfo(Variant::FLOAT, prefix + PNAME("angular_equilibrium_point"), PROPERTY_HINT_NONE, "radians_as_degrees"));
```

> The other question is a query again regarding the topic, and it's whether there are plans to add the missing motor functions and additional functions (softness, restitution, damping, ERP) to the 6DOFJoint of PhysicalBone3D at some point.

Which motor functions are you referring to? All the motor stuff should be supported as far as I know.

Attempts [were made](https://github.com/godot-jolt/godot-jolt/pull/430) at supporting the soft limit stuff (softness, restitution, damping, ERP), but in the end there was no good way of mapping it to the parameters that Jolt expects. So no, they will probably never be supported.

You can use the substitute joints instead (`JoltGeneric6DOFJoint3D`, etc.) if you need soft limits, but you won't be able to use them from `PhysicalBone3D` though, since it's hardcoded to only use the default set of joints.
